ext/standard/mail.c

```diff
@@ -281,6 +320,11 @@ PHPAPI int php_mail(char *to, char *subject, char *message, char *headers, char
 		efree(f);
 	}
 
+	if (hdr && php_mail_detect_multiple_crlf(hdr)) {
+		php_error_docref(NULL TSRMLS_CC, E_WARNING, "Multiple or malformed newlines found in additional_header");
+		MAIL_RET(0);
```

Comment:
@smalyshev I sent mail privately, but I'll leave comment here also.
I don't mind reverting this PR for now. We may consider how to deal with abuse of "string" $additional_header parameter for new releases.